README.md

```diff
@@ -186,3 +186,97 @@ repository environment variables.
 
 Travis already have a provider for deploying to bintray, but it is extremely
 inconvenient as it requires to produce one json file per file to upload.
+
+
+Building D1/2 projects
+----------------------
+
+Beaver provides some facilities to build D1 projects that are compatible with D2
+easily.
+
+This assumes you are using:
+
+- [MakD](https://github.com/sociomantic-tsunami/makd) (it might work for
+  other build systems that use Make though).
+- [Cachalot](https://github.com/sociomantic-tsunami/makd) `dlang` docker images.
+
+There are 2 main possible setups:
+
+1. Use the DMD privided by default by the image for some branch (dmd1,
+   dmd-transitional or dmd). This method is only encouraged for unstable,
+   bleeding-edge projects.
+
+2. Specify a particular DMD version. Recommended specially for libraries that
+   need to keep compatibility with multiple DMD versions, even among major
+   branches.
+
+Both approaches can be combined with Travis matrix builds, and in the following
+examples, matrix will always be used because is the more general case, but to do
+single builds just move the environment variables definition from `matrix:` to
+`global:` and that's it.
+
+Here is an example for case 1:
+
+```yml
+env:
+    matrix:
+        - DMD=dmd1
+        - DMD=dmd-transitional
+
+install: beaver dlang install
+
+script: beaver dlang make
+```
+
+`beaver dlang install` accepts arbitrary arguments that will be forwarded
+directly to the `beaver docker build` call.
+
+`beaver dlang make` by default runs the targets `test`, but you can override it
+by providing arguments. These arguments will be forwarded to `make` directly if
+provided. This command will **always** first run `make d2conv` if there is a D2
+compiler specified.
+
+For example, to get verbose output for some PR for debugging reasons, you could
+use `beaver dlang make V=1 test`.
+
+To specify a version explicitly (case 2) you can use the same commands, but just
+specify the version in the `$DMD` variable:
+
+```yml
+env:
+    matrix:
+        - DMD=1.079.0
+        - DMD=1.080.0
+        - DMD=2.070.2.s10
+        - DMD=2.070.2.s12
+        - DMD=2.074.0-0
+```
+
+In this case beaver will find out which is the package name to install and
+install that specific version. The rest of the Dockerfile stays the same.
+
+Then you'll also need to manage the installation of the package itself when
+building the docker images. For this, you need to specify an
+[argument](https://docs.docker.com/engine/reference/builder/#arg) to pass to the
+image.
+
+Here is an example `Dockerfile`:
+
+```dockerfile
+FROM sociomantictsunami/dlang:xenial-v2
+ARG DMD_PKG # This will be passed automatically by `beaver dlang install`
```

Comment:
Perhaps:

```
ARG DMD_PKG # This is calculated by beaver and it will be passed automatically by `beaver dlang install`
```

or something similar. My point was that this is not something that user need to provide (it should be not defined anywhere), we just need to make sure it's passed to `docker build`.